### ISSUE TITLE
`spec/integ/sliding-sync.spec.ts` tests are not independent

### ISSUE THREAD

**Issue by MadLittleMods** [matrix-org/matrix-js-sdk#2975]
We only assign `slidingSync` in a few of the [`spec/integ/sliding-sync.spec.ts`](https://github.com/matrix-org/matrix-js-sdk/blob/193c38523c6304c86e56d6bdc5089ece26ad34da/spec/integ/sliding-sync.spec.ts) tests which means if you try to focus a specific test that doesn't have the assignment, it will run into null pointer exceptions.

For a specific example see (`slidingSync` usage without assignment),

https://github.com/matrix-org/matrix-js-sdk/blob/193c38523c6304c86e56d6bdc5089ece26ad34da/spec/integ/sliding-sync.spec.ts#L1147-L1156

**Comment by t3chguy** [comment]
Yeah this was done intentionally I believe in an effort to make the tests feel less boilerplatey and faster, it does go against Cypress best practices though
